lib/rules/no-misleading-character-class.js

```diff
@@ -320,12 +265,55 @@ module.exports = {
                             } else {
                                 foundKindMatches.set(kind, [...findCharacterSequences[kind](chars)]);
                             }
-
                         }
                     }
                 }
             });
 
+            let codeUnits = null;
+
+            /**
+             * Finds the report loc(s) for a range of matches.
+             * Only literals and expression-less templates generate granular errors.
+             * @param {Character[][]} matches Lists of individual characters being reported on.
+             * @returns {Location[]} locs for context.report.
+             * @see https://github.com/eslint/eslint/pull/17515
+             */
+            function getNodeReportLocations(matches) {
+                return matches.map(chars => {
+                    const firstIndex = chars[0].start;
+                    const lastIndex = chars.at(-1).end - 1;
+                    let start;
+                    let end;
+
+                    if (node.type === "TemplateLiteral") {
+                        const source = sourceCode.getText(node);
+                        const offset = node.range[0];
+
+                        codeUnits ??= parseTemplateToken(source);
+                        start = offset + codeUnits[firstIndex].start;
+                        end = offset + codeUnits[lastIndex].end;
+                    } else if (typeof node.value === "string") { // String Literal
+                        const source = node.raw;
+                        const offset = node.range[0];
+
+                        codeUnits ??= parseStringLiteral(source);
+                        start = offset + codeUnits[firstIndex].start;
+                        end = offset + codeUnits[lastIndex].end;
+                    } else { // RegExp Literal
```

Comment:
else can be anything, for example:

![image](https://github.com/eslint/eslint/assets/44349756/46de832b-819b-4518-ba1b-f81c0c1e8e7f)

Reply:
Fixed in 7bb7e55f232840cfe564f11ce44a3f79a137b260.